events/events.go

```diff
@@ -31,9 +34,100 @@ type EventListener interface {
 	HandleEvent(event Event)
```

Comment:
I'm always confused that this isn't called EventHandler. Is there a way we can change it?